packages/server/src/utils/workflow.ts

```diff
@@ -158,9 +164,51 @@ function appendEventDataToAnnotations(
     downloadLlamaCloudFilesFromNodes(sourceNodes); // download files in background
   }
 
+  checkComponentAvailability(transformedEvent, componentsDir);
+
   dataStream.appendMessageAnnotation(transformedEvent.data as JSONValue);
 }
 
+function checkComponentAvailability(
```

Comment:
better move this concern to packages/server/src/handlers/components.ts?

Reply:
we could also move these warning logs to the FE as the frontend knows that `"sources", "agent"` are events with special renderers